### PR TITLE
Use xopen to automatically detect gzipped files.

### PR DESCRIPTION
**Pull request by gaberoo**
Small edit in `run.py` to read gzipped contig files.

### CODE REVIEW

**Comment by jakobnissen** [comment]
Dear @gaberoo 

Thank you for your contribution!
Vamb actually already contains a mechanism for automatic detection of gzipped contig files. See the `Reader` object in `vambtools.py`. I had simply forgotten to use it instead of the ordinary `open` when running Vamb from command line. In the `dev` version, the one I regularly add content to, this mistake has already been amended ([commit 397081)](https://github.com/jakobnissen/vamb/commit/397081e6447f42984bde00cfcecf170fb49e9dc4).

I will decline this PR, because it adds an unneeded dependency. Instead, I have added bzip2 and LZMA compression compatibility to the existing `Reader`, as this is offered by `xopen`. As usual, I will make this change on the `dev` branch, which is expected to merge into the master branch in less than one month.

By the way, if you're interested in using Vamb, I recommend checking out the next release due before the end of March (which will either be called 1.1.0 or 2.0), which contains quite a lot of changes, including a new and improved clustering algorithm, a new benchmarking algorithm, and tons of small quality-of-life improvements.

**Comment by gaberoo** [comment]
Dear Jakob,

Excellent! However you prefer is best.

With respect to the new clustering algorithm — is this already in the Dev branch? Because of the project timeline, I doubt I will be returning to the binning in a month's time.

**Comment by jakobnissen** [comment]
Yes, all the changes are in the dev branch. I feel obliged to warn you, however, that the `dev` branch is highly unstable and gets updated regularly.

Here's the current version of the change log compared to `master` branch:

### Highlights
* Vamb should now be installed using pip.
* The estimation of RPKM is now more accurate.
* Vamb now allows encoding of single-sample datasets.
* Clustering algorithm fundamentally changed.
* Benchmarking algorithm fundamentally changed.

### Now installable via pip
* Binaries are no longer part of the package. These are instead automatically compiled using Cython upon installation.
* Installing automatically installs Numpy, pysam and PyTorch if not installed already.
* Installation enforces Python >= 3.5.
* Installing via pip now adds the module to your $PATH so it can more easily be used from command line.

### Changes to RPKM estimation
* A BAM file with no reads now returns a depth of all zeros instead of all NaNs.
* The BAM parser no longer attempts to detect and filter away duplicate alignments erroneously produced by some aligners.
* Reads with a mapping mate are now consistently counted as half a read when estimating RPKM.
* Reads with N total subjects, where the secondary alignments are marked by ‘XA’ count as 1/N instead of 1 towards each subject.
* All alignments not marked as primary are now skipped.
* Minimum alignment score now works as expected when set below 1. Set to None in order to disable the filtering.
* Default minimum alignment score is now None instead of 50.
* Exceptions raised in subprocesses now propagate properly to the parent process and ends Vamb.

### Vamb now allows encoding of single-sample datasets
* When encoding a dataset with a single sample, the abundance loss uses sum of squared error, and does not softmax.
* Vamb still performs poorly on single-sample datasets compared to other binners.

### New clustering algorithm
* Now uses cosine distance instead of Pearson distance.
* Clustering threshold is estimated for each cluster while running instead of beforehand.
* The new approach is significantly slower, but more accurate.
* cluster.read_clusters now doesn’t crash when reading an empty line.
* Inputs are now shuffled in-place to reduce memory consumption.

### New benchmarking algorithm
* Now calculates recall and precision based on the total number of basepairs of the underlying genome covered by contigs in the bins.
* Changed the interface and required input files.
* Now accepts non-disjoint bins

### Other changes
* Vamb can now take TNF and RPKM inputs as precalculated .npz files.
* Contig parser now checks that the header contains no tab characters.
* Vamb now transparent accepts FASTA files zipped with gzip, bzip2 and LZMA.
* The mask is now saved in the output directory as mask.npz.
* The order of the columns in the RPKM array is now printed in the log file.
* Added wishlist.md to the repostory with future possible changes to Vamb.
